pkg/bootstrapper/bootstrapper_test.go

```diff
@@ -245,7 +244,11 @@ func TestCloudConfNotPresent(t *testing.T) {
 	err = wnb.parseIgnitionFileContents([]byte(ignitionContents), map[string]fileTranslation{})
 	assert.NoError(t, err, "error parsing ignition file contents")
 
-	_, err = os.Stat(path.Join(dir, "cloud.conf"))
+	cloudConfPath := filepath.Join(dir, "cloud.conf")
+	// Check file path was correctly generated for cloud.conf before checking its existence
+	assert.Equalf(t, string(os.PathSeparator), string(cloudConfPath[len(dir):][0]),
+		"Path not correctly appended for cloud.conf")
+	_, err = os.Stat(cloudConfPath)
```

Comment:
Here you are unit testing if `filepath.Join()` is doing the right thing. Your test needs to look at something that our code is generating and ensure that the correct path separator is being used.